libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -200,7 +204,13 @@ std::string IRGenerator::generate(
 	// Do not register immutables to avoid assignment.
 	t("DeployedObject", IRNames::deployedObject(_contract));
 	t("sourceLocationCommentDeployed", dispenseLocationComment(_contract));
-	t("library_address", IRNames::libraryAddressImmutable());
+	auto const eof = m_context.eofVersion().has_value();
+	t("eof", eof);
+	if (!eof)
+		t("library_address", IRNames::libraryAddressImmutable());
+	else
+		t("library_address_offset", std::to_string(CompilerUtils::generalPurposeMemoryStart));
```

Comment:
`generalPurposeMemoryStart` is the address in memory, but you're loading from aux data, so I think this is wrong. It should be just `0`:

```suggestion
		t("library_address_offset", 0);
```

You could actually put that directly in the template.

Reply:
done